### PR TITLE
[EAM-980] Open sourcing

### PR DESCRIPTION
**Pull request by pedroamarocern**

### CODE REVIEW

File: src/main/java/ch/cern/eam/wshub/core/services/equipment/entities/Location.java [hunk 1]
```diff
@@ -63,6 +63,9 @@
 	@Transient
 	private String hierarchyLocationCode;
 
+	@Transient
+	private String hierarchyLocationDesc;
```
Comment (BoykoBorisov) [inline src/main/java/ch/cern/eam/wshub/core/services/equipment/entities/Location.java:67]: Test updating Location hierarchy - Passed.

File: src/main/java/ch/cern/eam/wshub/core/services/material/entities/Part.java [hunk 2]
```diff
@@ -42,6 +42,10 @@
 	@InforField(xpath = "PARTID/DESCRIPTION")
 	private String description;
 
+	@Column(name = "PAR_ORG")
+	@InforField(xpath = "PARTID/ORGANIZATIONID/ORGANIZATIONCODE")
```
Comment (BoykoBorisov) [inline src/main/java/ch/cern/eam/wshub/core/services/material/entities/Part.java:46]: Test CRUD and associating part with asset - Passed on TEST.

File: src/main/java/ch/cern/eam/wshub/core/services/workorders/entities/WorkOrder.java [hunk 3]
```diff
@@ -69,13 +75,18 @@
 	@Transient
 	@InforField(xpath = "DEPARTMENTID/DESCRIPTION", readOnly = true)
 	private String departmentDesc;
+	@InforField(xpath = "DEPARTMENTID/ORGANIZATIONID/ORGANIZATIONCODE")
```
Comment (BoykoBorisov) [inline src/main/java/ch/cern/eam/wshub/core/services/workorders/entities/WorkOrder.java:78]: Test updating department - Passed on TEST.

File: src/main/java/ch/cern/eam/wshub/core/services/workorders/entities/WorkOrder.java [hunk 4]
```diff
@@ -69,13 +75,18 @@
...
+	private String departmentOrganization;
 	//
 	@Column(name = "EVT_OBJECT")
 	@InforField(xpath = "EQUIPMENTID/EQUIPMENTCODE")
 	private String equipmentCode;
 	@Transient
 	@InforField(xpath = "EQUIPMENTID/DESCRIPTION", readOnly = true)
 	private String equipmentDesc;
+	@Column(name = "EVT_OBJECT_ORG")
+	@InforField(xpath = "EQUIPMENTID/ORGANIZATIONID/ORGANIZATIONCODE")
```
Comment (BoykoBorisov) [inline src/main/java/ch/cern/eam/wshub/core/services/workorders/entities/WorkOrder.java:88]: Test updating equipment - Passed on Test.

File: src/main/java/ch/cern/eam/wshub/core/services/workorders/entities/WorkOrder.java [hunk 5]
```diff
@@ -93,6 +104,9 @@
 	@Transient
 	@InforField(xpath = "LOCATIONID/DESCRIPTION", readOnly = true)
 	private String locationDesc;
+	@Transient
+	@InforField(xpath = "LOCATIONID/ORGANIZATIONID/ORGANIZATIONCODE", nullifyParentLevel = 2)
```
Comment (BoykoBorisov) [inline src/main/java/ch/cern/eam/wshub/core/services/workorders/entities/WorkOrder.java:108]: Test updating location - Passed on TEST.

File: src/main/java/ch/cern/eam/wshub/core/services/workorders/impl/WorkOrderMiscServiceImpl.java [hunk 6]
```diff
@@ -41,12 +41,9 @@ public WorkOrderMiscServiceImpl(ApplicationData applicationData, Tools tools, In
 	}
 
 	public String createMeterReading(InforContext context, ch.cern.eam.wshub.core.services.workorders.entities.MeterReading meterReadingParam) throws InforException {
-
 		// Handling the normal case for meter reading
 		MeterReading meterreadinginfor = new MeterReading();
-
-		meterreadinginfor.setUSAGEUOMID(new UOMID_Type());
-		meterreadinginfor.getUSAGEUOMID().setUOMCODE(meterReadingParam.getUOM());
+		tools.getInforFieldTools().transformWSHubObject(meterreadinginfor, meterReadingParam, context);
```
Comment (BoykoBorisov) [inline src/main/java/ch/cern/eam/wshub/core/services/workorders/impl/WorkOrderMiscServiceImpl.java:46]: Test updating meter reading - Passed on TEST.

**Review by skamv — APPROVED** [review]

**Pull request merged**